extensions/amp-story/1.0/amp-story.js

```diff
@@ -1277,6 +1280,20 @@ export class AmpStory extends AMP.BaseElement {
       return Promise.resolve();
     }
 
+    if (
+      isExperimentOn(this.win, 'amp-story-paywall-exp') &&
+      targetPage.isPaywallProtected()
+    ) {
+      if (this.navigateToPageAfterSubscriptionsAreGranted_) {
```

Comment:
You could check this with `storeService_.get(StoreProperty.SUBSCRIPTIONS_DIALOG_STATE)` and not worry about storing the page yet. There's some logic regarding the page that it would make sense to keep out of this PR (eg: I believe you don't need to store the blocked page at all, given that the active page would be the one that holds this value). If you replace this line and remove the lines for `navigateToPageAfterSubscription...` you'll make this PR more condensed

Reply:
SG. Sure let's see if that works in later PRs.